src/core/storage/backend/memory-backend.ts

```
@@ -0,0 +1,125 @@
+/**
+ * In-memory storage backend for development and testing.
+ * Implements both CacheBackend and DatabaseBackend interfaces.
+ * Data is lost when the process restarts.
+ */
+export class MemoryBackend implements CacheBackend, DatabaseBackend {
+    private data = new Map<string, any>();
+    private lists = new Map<string, any[]>();
+    private ttls = new Map<string, number>();
+    private connected = false;
+
+    constructor() {}
```

Biome (1.9.4)

[error] This constructor is unnecessary.

Unsafe fix: Remove the unnecessary constructor.


(lint/complexity/noUselessConstructor)